contracts/SeriesIndexer.sol

```diff
@@ -0,0 +1,792 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/structs/BitMaps.sol";
+
+/**
+ * @title SeriesIndexer
+ * @dev A contract for managing series and artists in a collaborative art platform.
+ *      This contract allows artists to create series, manage co-artists, and control owner rights.
+ */
+contract SeriesIndexer is Ownable {
+    using BitMaps for BitMaps.BitMap;
+
+    // ============ Custom Errors ============
+
+    error OwnerRightsRevokedForThisSeries(uint256 seriesID);
+    error SeriesDoesNotExistError(uint256 seriesID);
+    error CallerNotASeriesArtistError(uint256 seriesID, address caller);
+    error NoArtistsProvidedError();
+    error NoArtistsForSeriesError();
+    error InvalidNewAddressError(address newAddress);
+    error AddressAlreadyAssignedError(address newAddress);
+    error InvalidArtistIDError(uint256 artistID);
+    error NotAuthorizedError(address caller);
+    error ArtistRevokedOwnerRightsError(address artistAddr);
+    error NoProposalExistsError(uint256 seriesID, address proposedArtistAddr);
+    error NotAPendingProposalError(uint256 seriesID, address caller);
+    error ZeroAddressNotAllowedError();
+    error NoSeriesDataError();
+    error BatchSizeTooLargeError(uint256 size);
+    error ArrayLengthMismatchError(uint256 lenArtists, uint256 lenMetas, uint256 lenTokens);
+    error AlreadyProposedError(uint256 seriesID, address proposedArtistAddr);
+    error ArtistAlreadyInSeriesError(uint256 seriesID, address proposedArtistAddr);
+    error NotAnArtistError(address caller);
+    error EmptyMetadataURIError();
+    error EmptyTokenMapURIError();
+
+    // ============ Structs ============
+
+    struct Series {
+        string metadataURI;              // IPFS hash or similar identifier for series metadata
+        string contractTokenDataURI;     // Token-related data for the series
+        uint256[] artistIDs;          // List of artists associated with this series
+        uint256[] pendingCoArtists;   // List of pending co-artists
```

Comment:
Maintaining a pending artist list within a series is not a good choice to me. The reason is the pending artist is just a temporary data acts as an intermediate stage for artist proposal. In other words, it doesn't really relate to the series. The series should have artist, metadata and token data, that's what we need for a series. When query the series from contract, we don't care about the co-artist since it's not official one. 
Instead, I'd suggest maintain a pending co-artist as separate data, out of the series.